monri/src/main/java/com/monri/android/view/CardNumberEditText.java

```diff
@@ -176,7 +178,7 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                     updateCardBrandFromNumber(s.toString());
                 }
 
-                if (start > 16) {
+                if (start > 21) {
```

Comment:
Hm, why 21? If I follow logic 16 is standard visa card formatted:  `**** **** **** ****`.

Can you explain what's really happening here?

Reply:
If visa have 19 digits then it's something like **** **** **** **** ***, and that new part start from index 21

Reply:
So you are basically interested in index 22? Because if I'm typing last digit of 19 digits long pan that's character at index 22. Got it.